api/v1beta1/azuremachine_validation.go

```diff
@@ -234,6 +238,16 @@ func validateManagedDisk(m *ManagedDiskParameters, fieldPath *field.Path, isOSDi
 
 	if m != nil {
 		allErrs = append(allErrs, validateStorageAccountType(m.StorageAccountType, fieldPath.Child("StorageAccountType"), isOSDisk)...)
+
+		if isOSDisk && m.SecurityProfile != nil && m.SecurityProfile.DiskEncryptionSet != nil {
+			if m.SecurityProfile.SecurityEncryptionType != SecurityEncryptionTypeDiskWithVMGuestState {
```

Comment:
Do you think we could add a comment here explaining this validation logic for a quick review ? 
Would be great to add link to the validation rules too. 

For instance, I found referring to https://learn.microsoft.com/en-us/rest/api/compute/virtual-machines/create-or-update?tabs=HTTP#securityencryptiontypes to be helpful understanding this validation.

Reply:
I tried to add useful comments to the validations, but I'm afraid that for most of them there is no detailed Azure REST API doc link that we could use. The best way I managed to verify them was via trial and error, i.e. creating VMs with different configuration, after checking out the respective [Azure Terraform provider documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine#security_encryption_type).